nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/impl/TestTreeCommentManager.java

```diff
@@ -276,6 +278,28 @@ public void shouldCreateCommentAndReplies() {
         assertEquals(doc.getRef(), topLevelCommentAncestor);
     }
 
+    @Test
+    public void shouldIndicateTheRelevantParent() {
+        DocumentModel doc = createDocumentModel("yetAnotherFile");
+        try (CapturingEventListener listener = new CapturingEventListener(COMMENT_ADDED)) {
+            // Create a comment
+            Comment commentToCreate = createSampleComment(doc.getId());
+            Comment createdComment = createAndCheckComment(session, doc, commentToCreate, 1);
+            assertEquals(1, listener.getCapturedEventCount(COMMENT_ADDED));
+            Event event = listener.getCapturedEvents().get(0);
+            DocumentModel parent = (DocumentModel) event.getContext().getProperty(COMMENTED_DOCUMENT);
+            assertEquals(doc.getId(), parent.getId());
+            // Create a reply
+            Comment firstReply = createSampleComment(createdComment.getId());
+            firstReply = createAndCheckComment(session, session.getDocument(new IdRef(createdComment.getId())),
+                    firstReply, 2);
+            assertEquals(2, listener.getCapturedEventCount(COMMENT_ADDED));
+            event = listener.getCapturedEvents().get(1);
+            parent = (DocumentModel) event.getContext().getProperty(COMMENTED_DOCUMENT);
+            assertEquals(doc.getId(), parent.getId());
+        }
+    }
+
     @Test
```

Comment:
you should add this test on the `AbstractTestCommentManager` to handle the `TreeCommentManager` and `PropertyCommentManager` -> we should test the both services.

Reply:
Removed as rebasing on NXP-28253